napi.h

```diff
@@ -1202,8 +1205,30 @@ class Object::iterator {
 };
 #endif  // NODE_ADDON_API_CPP_EXCEPTIONS
 
+class ArrayBufferLike : public Object {
```

Comment:
Usually we use the name `XxxLike` when it describes a shape of a protocol. For ArrayBuffer and SharedArrayBuffer, this is not the case. The `Napi::ArrayBuffer` and `Napi::SharedArrayBuffer` do not work with a JavaScript mocked ArrayBuffer-like objects.

I think a better name can be `ArrayBufferBase`. Or just duplicate these two methods as in JavaScript, `ArrayBuffer` and `SharedArrayBuffer` have no common ancestry.

Reply:
Addressed in https://github.com/nodejs/node-addon-api/pull/1688/commits/f861d36e5aba8b3d84f78f08a3399fea1c84128e

Reply:
I went with the approach of:

> Or just duplicate these two methods as in JavaScript, ArrayBuffer and SharedArrayBuffer have no common ancestry.